assets/js/googlesitekit/widgets/datastore/widgets.js

```diff
@@ -0,0 +1,258 @@
+/**
+ * core/widgets data store: widgets info.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from './constants';
+
+const { commonActions, createRegistrySelector } = Data;
+
+/**
+ * Store our widget components by registry, then by widget `slug`. We do this because
+ * we can't store React components in our data store.
+ *
+ * @private
+ * @since n.e.x.t
+ */
+export const WidgetComponents = {};
+
+const ASSIGN_WIDGET = 'ASSIGN_WIDGET';
+const REGISTER_WIDGET = 'REGISTER_WIDGET';
+const SET_WIDGET_COMPONENT_KEY = 'SET_WIDGET_COMPONENT_KEY';
+
+export const INITIAL_STATE = {
+	registryKey: undefined,
+	widgets: {},
+};
+
+export const actions = {
+	/**
+	 * Assigns an existing widget (by slug) to a widget area(s).
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param  {string}         slug      Widget slug.
+	 * @param  {(string|Array)} areaSlugs Widget Area slug(s).
+	 * @return {Object}                   Redux-style action.
+	 */
+	assignWidget( slug, areaSlugs ) {
+		let areaSlugsAsArray;
+		if ( typeof areaSlugs === 'string' ) {
+			areaSlugsAsArray = [ areaSlugs ];
+		} else {
+			areaSlugsAsArray = [ ...areaSlugs ];
+		}
+
+		return {
+			payload: { slug, areaSlugs: areaSlugsAsArray },
+			type: ASSIGN_WIDGET,
+		};
+	},
+
+	/**
+	 * Register a widget with a given slug and settings.
+	 *
+	 * @since n.e.x.t
+	 * @param {string} slug           Widget's slug.
+	 * @param {Object} settings       Widget's settings.
+	 * @return {Object}          Redux-style action.
```

Comment:
Can you add documentation for all the `settings` arguments here? `component` should be required, `priority` (default `10`) and `width` (default `'quarter'`) optional.